tests/struct.rs

```diff
@@ -468,3 +468,48 @@ fn default_bits() {
     let default = ArrayTupleDefault::default();
     assert_eq!(default, ArrayTupleDefault::from(u34::new(0b1000_1000_1000_10_10_1000_01000_1000_01000)));
 }
+
+// quick 'n dirty -- consider something more robust if more manual implementations are needed
+#[cfg(not(feature = "nightly"))]
+macro_rules! impl_from (
+    ($($generic:ident),*; $from_ty:ty => $to_ty:ty; |$name:ident| $expr:expr) => {
+        impl<$($generic),*> From<$from_ty> for $to_ty {
+            fn from($name: $from_ty) -> $to_ty {
+                $expr
+            }
+        }
+    }
+);
+#[cfg(feature = "nightly")]
+macro_rules! impl_from (
+    ($($generic:ident),*; $from_ty:ty => $to_ty:ty; |$name:ident| $expr:expr) => {
+        impl<$($generic),*> const From<$from_ty> for $to_ty {
+            fn from($name: $from_ty) -> $to_ty {
+                $expr
+            }
+        }
+    }
+);
+
+// do we want to be able to derive automatically on structs where the user has specified phantomdata?
```

Comment:
I think that would be done by removing the PhantomData field in `bitsize_internal` and adding it back at the end. We should do that, in a separate PR though.